src/virtualenv/seed/wheels/bundle.py

```diff
@@ -14,7 +14,7 @@ def from_bundle(distribution, version, for_py_version, search_dirs, app_data, do
 
     if version != Version.embed:
         # 2. check if we have upgraded embed
-        if app_data.can_update:
+        if of_version is None and app_data.can_update:
             wheel = periodic_update(distribution, for_py_version, wheel, search_dirs, app_data, do_periodic_update, env)
```

Comment:
Another option is to pass `of_version` to `periodic_update` as is done for `load_embed_wheel` and `from_dir`.

Reply:
Would be nice to do that instead to keep the same design.